### PR TITLE
[IA-4855] New delete result endpoint for storage container

### PR DESCRIPTION
**Pull request by LizBaldo**
JIRA Ticket: https://broadworkbench.atlassian.net/browse/IA-4855

This should be the last of the ASYNC result endpoints that Leo needs to properly poll resource creation / deletion. I tried to be a good scout and add a little unit test file to cover creation, cloning and deletion of the storage container, but let me know if I missed anything.

Tested on my BEE and things are working as expected 👍  (aka creating a container, sending an async delete command, and then monitoring via the new delete-result endpoint using the delete job id)

One thing I noticed is that there isn't a `GET` endpoint for storage containers, would it be something useful to add or was that left out intentionally?

### CODE REVIEW

**Review by marctalbott — APPROVED** [review]
Couple of extra files to remove, but the changes themselves look good.

As for the `getStorageContainer` API, I don't know the reason it has not been implemented, but I doubt it was intentional. You can use the more generic [getResource](https://workspace.dsde-dev.broadinstitute.org/swagger-ui.html#/Resource/getResource) API if you'd like, or if you would like to implement it that would be okay with us.

File: service/hs_err_pid89591.log [hunk 1]
```diff
@@ -0,0 +1,1472 @@
+#
```
[the remark this answers is not in the document]
Comment (LizBaldo) [reply service/hs_err_pid89591.log:1]: oops, thanks for the catch, will remove :)

**Review by cahrens — APPROVED** [review]
Thank you for doing this!

File: service/src/test/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiControllerAzureDatabaseTest.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public void createDatabase() throws Exception {
                 commonFields,
                 Region.US_SOUTH_CENTRAL.name(),
                 USER_REQUEST,
-                WsmResourceType.CONTROLLED_AZURE_VM));
+                WsmResourceType.CONTROLLED_AZURE_DATABASE));
```
Comment (lucymcnatt) [inline service/src/test/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiControllerAzureDatabaseTest.java:91]: nice catches here on these tests

**Review by lucymcnatt — APPROVED** [review]

**Pull request merged**